parlai/agents/bert_ranker/bi_encoder_ranker.py

```diff
@@ -46,7 +46,7 @@ def __init__(self, opt, shared=None):
         super().__init__(opt, shared)
         # it's easier for now to use DataParallel when
         self.data_parallel = opt.get('data_parallel') and self.use_cuda
-        if self.data_parallel:
+        if self.data_parallel and shared is None:
             self.model = torch.nn.DataParallel(self.model)
         if is_distributed():
```

Comment:
this should probably go under the check for `if self.data_parallel`..

Reply:
?

Reply:
the `if_distributed()` call raises an error and says distributed and data_parallel aren't compatible. but it doesn't make sense to raise this error `if not self.data_parallel`...

Reply:
(it is not related to the change that @hadasah made, it's an error i just noticed)

Reply:
Created a new issue #2020.